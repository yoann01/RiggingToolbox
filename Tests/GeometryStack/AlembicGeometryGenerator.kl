
require RiggingToolbox;

operator entry(){

  AlembicGeometryGenerator alembicGenerator();
  alembicGenerator.setFilePath("${FABRIC_RIGGINGTOOLBOX_PATH}/Tests/GeometryStack/Resources/cubeTest.abc");

  EvalContext context();
  GeometrySet geomSet();
  alembicGenerator.evaluate(context, geomSet);

  report("NumGeometries:" + geomSet.size);
  for(Integer i=0; i<geomSet.size; i++){
    report(geomSet.get(i).getDesc(false, false));
  }
}
