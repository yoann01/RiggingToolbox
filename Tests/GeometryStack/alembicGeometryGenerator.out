Importing:D:/Projects/FabricEngineInc/RiggingToolbox/Tests/GeometryStack/Resources/cubeTest.abc
NumGeometries:1
Mesh: pointCount: 8 polygonCount: 6 nbAttributeVectors: 8
  Points (adjacent polygons as 'polygon.neighborPolygonIndex',
          borders as '|', closed wing starts as '<<'):
    0: 3 polygons:  <<2.0, 0.0, 1.0
    1: 3 polygons:  <<3.0, 1.1, 0.3
    2: 3 polygons:  <<4.0, 0.1, 2.3
    3: 3 polygons:  <<4.3, 3.1, 0.2
    4: 3 polygons:  <<5.0, 2.1, 1.3
    5: 3 polygons:  <<5.1, 1.2, 3.3
    6: 3 polygons:  <<5.3, 4.1, 2.2
    7: 3 polygons:  <<5.2, 3.2, 4.2
  Polygons (connected points as 'point.polygonPointIndex', borders as '|'):
    0: 4 points: 0.1, 2.1, 3.2, 1.2
    1: 4 points: 0.2, 1.1, 5.1, 4.2
    2: 4 points: 0.0, 4.1, 6.2, 2.2
    3: 4 points: 1.0, 3.1, 7.1, 5.2
    4: 4 points: 2.0, 6.1, 7.2, 3.0
    5: 4 points: 4.0, 5.0, 7.0, 6.0

